src/App.js

```diff
@@ -1,16 +1,23 @@
-import { Routes, Route } from "react-router-dom";
-import React from "react";
+import React, { useEffect } from "react";
+import { Routes, Route, useLocation, useSearchParams } from "react-router-dom";
 
 import SideNav from "./components/side-nav/SideNav";
 import Header from "./components/header/Header";
 import Home from "./pages/home/Home";
 import Search from "./pages/search/Search";
 import Login from "./pages/login/Login";
-import { useLocation } from "react-router-dom";
 
 function App() {
```

Comment:
'functional' component is just a simple JS function; it accepts the data in the form of props and returns the react element. Whereas the class component will be created using the class keyword, and it extends the React.